hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -789,6 +809,14 @@ public Long getMaxMemoryPerPartitionMerge() {
     private boolean isConsistencyGuardSet = false;
     private boolean isCallbackConfigSet = false;
 
+    public Builder() {
+      this(EngineType.SPARK);
+    }
+
+    public Builder(EngineType engineType) {
```

Comment:
how about adding a method `withEngineType()` instead of making it a `Builder` constructor param?

Reply:
> how about adding a method `withEngineType()` instead of making it a `Builder` constructor param?

done